### PR TITLE
Settings api

### PR DESCRIPTION
**Pull request by ibolton336**
Add api support for mvn,svn, & git settings configuration from the repositories nav section.

### CODE REVIEW

File: pkg/client/src/app/pages/repositories/Mvn.tsx [hunk 1]
```diff
@@ -36,7 +104,7 @@ export const RepositoriesMvn: React.FunctionComponent = () => {
       </PageSection>
       <PageSection>
         <Card>
-          <CardBody>
+          {/* <CardBody>
```
Comment (gildub) [inline pkg/client/src/app/pages/repositories/Mvn.tsx:107]: Is this needed ?

File: pkg/client/src/app/pages/repositories/Mvn.tsx [hunk 2]
```diff
@@ -48,14 +116,14 @@ export const RepositoriesMvn: React.FunctionComponent = () => {
             <Button variant="link" isInline>
               Clear repository
             </Button>
-          </CardBody>
+          </CardBody> */}
           <CardBody>
```
Comment (gildub) [inline pkg/client/src/app/pages/repositories/Mvn.tsx:120]: Don't we need error handling here ?

File: pkg/client/src/app/pages/repositories/Svn.tsx [hunk 3]
```diff
@@ -35,7 +72,7 @@ export const RepositoriesSvn: React.FunctionComponent = () => {
               className="repo"
               label="Consume insecure Subversion repositories"
               aria-label="Insecure Subversion Repositories"
-              isChecked={isInsecure}
+              isChecked={svnInsecureSetting === true ? true : false}
               onChange={onChange}
             />
           </CardBody>
```
Comment (gildub) [inline pkg/client/src/app/pages/repositories/Svn.tsx:78]: Don't we need error handling here too ?

**Review by gildub — COMMENTED** [review]
Please see inline.
The rest.tsx file is becoming really difficult to follow don't you think it might be time to make it more modular?
Git.tsx, Svn.tsx and Mvn.tsx are very similar, where the first 2 ones are the same and seems good candidates for a share component.

File: pkg/client/src/app/api/rest.tsx [hunk 4]
```diff
@@ -753,6 +756,7 @@ export const getBulkCopyAssessment = (
 };
 
 const identityHeaders = { Accept: "application/json" };
+const settingsHeaders = { Accept: "application/json" };
```
[the remark this answers is not in the document]
Comment (ibolton336) [reply pkg/client/src/app/api/rest.tsx:759]: +! sounds good

File: pkg/client/src/app/pages/repositories/Mvn.tsx (hunk 1 above)
Comment (ibolton336) [reply pkg/client/src/app/pages/repositories/Mvn.tsx:107]: I will add a TODO for a stretch goal for this. As jeff mentioned.

File: pkg/client/src/app/pages/repositories/Mvn.tsx (hunk 2 above)
Comment (ibolton336) [reply pkg/client/src/app/pages/repositories/Mvn.tsx:120]: Good point

**Comment by ibolton336** [comment]
> 

I agree but I don't think we have the bandwidth to take on tech debt for now. Let's revisit when things calm down if that makes sense?

> Please see inline. The rest.tsx file is becoming really difficult to follow don't you think it might be time to make it more modular? Git.tsx, Svn.tsx and Mvn.tsx are very similar, where the first 2 ones are the same and seems good candidates for a share component.

**Review by gildub — APPROVED** [review]
LGTM, besides the technical debt we're adding here.

**Comment by ibolton336** [comment]
> LGTM, besides the technical debt we're adding here.

Not sure I added any here since those files were already in existence & not modular.

**Pull request merged**